dm/dm/master/scheduler/scheduler.go

```diff
@@ -806,6 +806,20 @@ func (s *Scheduler) AddSubTasks(latched bool, cfgs ...config.SubTaskConfig) erro
 		m[stage.Source] = stage
 	}
 
+	// 6. check if we can resolve load task by scheduling the source to the worker which has dump files on local disk.
+	for _, cfg := range newCfgs {
+		source := cfg.SourceID
+		worker, ok := s.bounds[source]
+		if !ok {
+			continue
+		}
+
+		if err2 := s.tryResolveLoadTask(worker.BaseInfo().Name, source); err2 != nil {
```

Comment:
`AddSubTasks` only hold RLock ,but when `tryResolveLoadTask`  trigger `transfer-source ` we need write lock